### PR TITLE
Update Azure Disk Driver V2 test k8s version and test triggers.

### PR DESCRIPTION
**Pull request by landreasyan**

### CODE REVIEW

File: config/jobs/kubernetes-sigs/azuredisk-csi-driver/azuredisk-csi-driver-v2-config.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ presubmits:
       testgrid-num-columns-recent: '30'
   - name: pull-azuredisk-csi-driver-unit-mainv2
     decorate: true
-    always_run: true
+    skip_if_only_changed: "^docs/|^site/|^\\.github/|\\.(md|adoc)$|^(README|LICENSE)$"
```
Comment (edreed) [inline config/jobs/kubernetes-sigs/azuredisk-csi-driver/azuredisk-csi-driver-v2-config.yaml:31]: I believe the triggers are already covered by #25725

Comment (landreasyan) [reply config/jobs/kubernetes-sigs/azuredisk-csi-driver/azuredisk-csi-driver-v2-config.yaml:31]: Oh, I didn't see that PR! Will rebase once it merges.

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**